pandas/indexes/numeric.py

```diff
@@ -19,6 +19,32 @@ class NumericIndex(Index):
     """
     _is_numeric_dtype = True
 
+    def __new__(cls, data=None, dtype=None, copy=False, name=None,
+                fastpath=False, **kwargs):
+
+        if fastpath:
+            return cls._simple_new(data, name=name)
+
+        # isscalar, generators handled in coerce_to_ndarray
+        data = cls._coerce_to_ndarray(data)
+
+        if issubclass(data.dtype.type, compat.string_types):
+            cls._string_data_error(data)
+
+        if copy or data.dtype != cls._default_dtype:
+            try:
+                subarr = np.array(data, cls._default_dtype, copy=copy)
+                assert((subarr == data) | np.isnan(subarr)).all()
```

Comment:
ok, will need several tests w.r.t. to whether a passed underlying array is actually copied or not (there are some already in ``indexes/tests/test_numeric.py/Numeric`` but have to make sure that these are correctly tested